File: ruby/fizzbuzz.rb
class FizzBuzz
    def fizz?(number)
        number % 3 == 0
    end

    def buzz?(number)
        number % 5 == 0
    end

    def take(number)
        if fizz?(number) and buzz?(number)
            "fizzbuzz"
        elsif fizz?(number)
            "fizz"
        elsif buzz?(number)
            "buzz"
        else
            number
        end
    end
end

File: R/fizzbuzz.R
fizzbuzz = function(n) {
  if(n %% 3 == 0 & n %% 5 == 0) return('fizzbuzz')
  else if (n %% 3 == 0) return('fizz')
  else if (n %% 5 == 0) return('buzz')

  n
}
